src/app/(sidebar)/smart-contracts/deploy-contract/page.tsx

```diff
@@ -176,88 +234,192 @@ export default function DeployContract() {
+        description:
+          "Upload transaction was created successfully. You can sign it next.",
+        type: "success",
+      });
+    }
+  }, [addFloatNotification, isUploadTxSuccess]);
+
+  // Show float notification when upload tx is signed
+  useEffect(() => {
+    if (signUploadSuccess) {
+      addFloatNotification({
+        id: "upload-sign-tx-success",
+        title: "Upload transaction signed",
+        description: `${signUploadSuccess}. You can submit it to the network next.`,
+        type: "success",
+      });
     }
-  }, [isUploadTxSuccess]);
+  }, [addFloatNotification, signUploadSuccess]);
 
+  // Show float notification when upload tx is submitted
   useEffect(() => {
     if (isSubmitUploadTxSuccess) {
-      setCurrentStep(4);
+      addFloatNotification({
+        id: "upload-submit-tx-success",
+        title: "Contract uploaded to the network",
+        description: `${selectedFile?.name} file has been uploaded successfully.`,
+        type: "success",
+      });
+
+      // Collapse the upload block on success
+      delayedAction({
+        action: () => {
+          setIsUploadExpanded(false);
+          setIsDeployExpanded(true);
+        },
+        delay: 300,
+      });
     }
-  }, [isSubmitUploadTxSuccess]);
+  }, [addFloatNotification, isSubmitUploadTxSuccess, selectedFile?.name]);
 
+  // ===========================================================================
+  // Deploy effects
+  // ===========================================================================
+
+  // Show float notification on deploy tx created success
   useEffect(() => {
     if (isDeployTxSuccess) {
-      setCurrentStep(5);
+      addFloatNotification({
+        id: "deploy-build-tx-success",
+        title: "Deploy transaction created",
+        description:
+          "Deploy transaction was created successfully. You can sign it next.",
+        type: "success",
+      });
     }
-  }, [isDeployTxSuccess]);
+  }, [addFloatNotification, isDeployTxSuccess]);
 
+  // Show float notification when deploy tx is signed
   useEffect(() => {
-    if (isSubmitDeployTxSuccess) {
-      setCurrentStep(0);
+    if (signDeploySuccess) {
+      addFloatNotification({
+        id: "deploy-sign-tx-success",
+        title: "Deploy transaction signed",
+        description: `${signDeploySuccess}. You can submit it to the network next.`,
+        type: "success",
+      });
     }
-  }, [isSubmitDeployTxSuccess]);
+  }, [addFloatNotification, signDeploySuccess]);
 
+  // Show float notification when deploy tx is submitted
   useEffect(() => {
-    if (selectedFile && currentStep > 0) {
-      const stepRef = getStepRef(currentStep);
-
-      if (stepRef?.current) {
-        scrollElIntoView(stepRef);
-      }
+    if (isSubmitDeployTxSuccess) {
+      addFloatNotification({
+        id: "deploy-submit-tx-success",
+        title: "Contract deployed!",
+        description: `${selectedFile?.name} file has been successfully deployed.`,
```

Comment:
<img width="402" height="117" alt="Image" src="https://github.com/user-attachments/assets/d7f422ce-3e46-4b43-8e91-ed31051dcbaa" /> can we wrap the text?

Reply:
> Could you please clarify what you mean by "wrap" here?

`successfully` is broken into two